### PR TITLE
Refactor ResourceBase to make the options sent to rest-client more configurable

### PR DESCRIPTION
**Pull request by drcapulet**
I need to be able to configure a proxy for rest-client on a per-client basis so I refactored this first to normalize between different request methods.

### CODE REVIEW

**Review by colin-garcia — APPROVED** [review]

**Comment by ShariqT** [comment]
Thanks for the interest in Lob. So, we are looking at the PR and it looks like this will add `body` content to the GET and DELETE methods. That's something that we are not really prepared to do at the moment. Can you explain what you want to be able to do a little bit more and how the `body` content will allow this?

File: lib/lob/resources/resource_base.rb [hunk 1]
```diff
@@ -71,16 +70,20 @@ def submit(request)
...
+
+        client_params = {
+          headers: headers,
+          method: request[:method],
+          url: url,
+        }
+        client_params[:body] = body if body
+
         begin
-          if method == :get || method == :delete
```
Comment (mwj8410) [inline lib/lob/resources/resource_base.rb:76]: Can we wrap the assignment to `client_params[:body]` with these 2 cases please?

GET and DELETE operations are intended by specification to only leverage the URI. Removing this case enables passing a body in these requests. Although it isn't specifically prohibited it the spec, it is intended to be functionally irrelevant.

Ref: [https://www.rfc-editor.org/rfc/rfc2616#section-9.3](https://www.rfc-editor.org/rfc/rfc2616#section-9.3)

**Review by mwj8410 — CHANGES_REQUESTED** [review]

**Comment by drcapulet** [comment]
Ah to clarify - the change isn't intending to set body for `GET` or `DELETE` and I can certainly add back that guard (though practically it'd be an issue with the caller anyways). As I noted in the PR description, I need to be able to pass rest-client a [proxy](https://github.com/rest-client/rest-client#proxy) and the `RestClient.get` variety methods don't support that so I decided to normalize things for the `RestClient::Request.execute` call.

**Review by mwj8410 — APPROVED** [review]
Perfect. Thank very much

**Comment by ShariqT** [comment]
Thanks for the contribution! We have merged this piece of code and are looking over the other PR you submitted as well.

**Pull request merged**